cli/cmd/project/clone.go

```diff
@@ -0,0 +1,153 @@
+			}
+
+			// get project
+			res, err := client.GetProject(cmd.Context(), &adminv1.GetProjectRequest{OrganizationName: ch.Org, Name: name})
+			if err != nil {
+				return err
+			}
+
+			if res.Project.ArchiveAssetId != "" {
+				return fmt.Errorf("project is not connected to a git repository, please redeploy the project to connect it to a git repository")
+			}
+
+			// check if dir is empty
+			empty, err := isDirAbsentOrEmpty(path)
+			if err != nil {
+				return err
+			}
+			if !empty {
+				ok, err := cmdutil.ConfirmPrompt(fmt.Sprintf("Directory %q is not empty. Do you want to overwrite it?", path), "", false)
+				if err != nil {
+					return err
+				}
+				if !ok {
+					return fmt.Errorf("directory %q is not empty", path)
+				}
+			}
+
+			// create directory, the go-git SDK does not create the directory
+			err = recreateDir(path)
+			if err != nil {
+				return fmt.Errorf("failed to create directory %q: %w", path, err)
+			}
+
+			// get config
+			config, err := cmdutil.NewGitHelper(client, ch.Org, name, path).FetchGitConfig(cmd.Context())
+			if err != nil {
+				return err
+			}
+
+			// clone repository
+			_, err = git.PlainCloneContext(cmd.Context(), path, false, &git.CloneOptions{
+				URL:           config.Remote,
+				Auth:          &githttp.BasicAuth{Username: config.Username, Password: config.Password},
+				ReferenceName: plumbing.NewBranchReferenceName(config.DefaultBranch),
+				SingleBranch:  true,
+			})
+			if err != nil {
+				return err
+			}
+
+			ch.Printf("Cloned project %q to %q\n", name, path)
+
+			// download variables
+			err = env.PullVars(cmd.Context(), ch, path, name, "prod", false)
+			if err != nil {
+				return fmt.Errorf("failed to download variables: %w", err)
+			}
+
+			// set rill cloud
+			err = dotrillcloud.SetAll(path, ch.AdminURL(), &dotrillcloud.Config{
+				ProjectID: res.Project.Id,
+			})
+			if err != nil {
+				return fmt.Errorf("failed to set rill cloud config: %w", err)
+			}
+
+			repo, _, err := cmdutil.RepoForProjectPath(path)
+			if err != nil {
+				return err
+			}
+
+			_, err = cmdutil.EnsureGitignoreHasDotRillCloud(cmd.Context(), repo)
+			if err != nil {
+				return fmt.Errorf("failed to update .gitignore: %w", err)
+			}
+			return nil
+		},
+	}
+
+	cloneCmd.Flags().SortFlags = false
+	cloneCmd.Flags().StringVar(&path, "path", "", "Project path to clone to")
+
+	return cloneCmd
+}
+
+func isDirAbsentOrEmpty(path string) (bool, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return true, nil
+		}
+		return false, err
+	}
+	defer f.Close()
+
+	// Read the directory entries
+	entries, err := f.Readdirnames(1)
+	if err != nil {
+		return false, err
+	}
+
+	// Check if the directory is empty
+	return len(entries) == 0, nil
+}
+
+func recreateDir(path string) error {
+	// Remove directory and its contents if exists
+	err := os.RemoveAll(path)
+	if err != nil {
+		// NOTE :: does not return an error if the directory does not exist
```

Comment:
nit: it's English, not C++ 😄 
```suggestion
		// NOTE: does not return an error if the directory does not exist
```